controller/dataplane/owned_deployment.go

```diff
@@ -271,6 +280,46 @@ func listOrReduceDataPlaneDeployments(
 	return false, &deployments[0], nil
 }
 
+func podTemplateSpecHasRestartAnnotation(template *corev1.PodTemplateSpec) (string, bool) {
+	if template == nil || template.Annotations == nil {
+		return "", false
+	}
+	v, ok := template.Annotations[restartAnnotationKey]
+	return v, ok && v != ""
+}
+
+// isRecentDeploymentRestart detects if a deployment is undergoing a recent restart operation.
+// It checks for the presence of the kubectl restart annotation and verifies if it's recent (within 5 minutes).
+// Returns true if a recent restart is detected, false otherwise, along with the restart timestamp if present.
+func isRecentDeploymentRestart(template *corev1.PodTemplateSpec, logger logr.Logger) (bool, string) {
```

Comment:
Minor nit:

I'd keep the order of the return values to be the same as in other functions: so

```suggestion
func isRecentDeploymentRestart(template *corev1.PodTemplateSpec, logger logr.Logger) (string, bool) {
```